harness/features/initialize.local.test.ts

```diff
@@ -0,0 +1,176 @@
+                    .reply(200, {})
+
+                const response = await createClient(url, clientId, 'invalid key')
+                const { exception } = await response.json()
+                await wait(500)
+
+                expect(exception).toEqual(
+                    'Invalid environment key provided. Please call initialize with a valid server environment key'
+                )
+                expect(scope.pendingMocks().length).toEqual(1)
+            })
+
+            it('initializes correctly on valid data', async () => {
+                const clientId = uuidv4()
+                scope
+                    .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+                    .reply(200, {})
+
+                const response = await createClient(
+                    url,
+                    clientId,
+                    sdkKey,
+                    { baseURLOverride: `${harnessBaseURL}/client/${clientId}` }
+                )
+                await wait(500)
+
+                const { message } = await response.json()
+                expect(message).toEqual('success')
+                expect(scope.isDone()).toBeTruthy()
+            })
+
+            it('calls initialize promise/callback when config is successfully retrieved', async () => {
+                const clientId = uuidv4()
+                const callbackSubdirectory = `/client/${clientId}/onClientInitialized`
+                scope
+                    .post(callbackSubdirectory, { message: `onClientInitialized was invoked on /client/${clientId}` })
+                    .matchHeader('Content-Type', 'application/json')
+                    .reply(204)
+                
+                await createClient(url, clientId, sdkKey)
+                await callOnClientInitialized(clientId, url, `${harnessBaseURL}${callbackSubdirectory}`)
+                await wait(500)
+
+                expect(scope.isDone()).toBeTruthy()
+            })
+
+            it('calls initialize promise/callback when config fails to be retrieved', async () => {
+                const clientId = uuidv4()
+                scope
+                    .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+                    .reply(404)
+
+                const callbackSubdirectory = `/client/${clientId}/onClientInitialized`
+                scope
+                    .post(callbackSubdirectory, { message: `onClientInitialized was invoked on /client/${clientId}` })
+                    .matchHeader('Content-Type', 'application/json')
+                    .reply(204)
+
+                await createClient(url, clientId, sdkKey, { baseURLOverride: `${harnessBaseURL}/client/${clientId}` })
+                await callOnClientInitialized(clientId, url, `${harnessBaseURL}${callbackSubdirectory}`)
+                await wait(500)
+
+                expect(scope.isDone()).toBeTruthy()
+            })
+
+            it('fetches config again after 10 seconds', async () => {
```

Comment:
debating if we even need this case since youre also testing it below, and 10 seconds is a long time to have to wait in a test suite. Thoughts?